examples/gcoap/gcoap_cli.c

```diff
@@ -314,12 +366,27 @@ int gcoap_cli_cmd(int argc, char **argv)
      */
     if (((argc == apos + 3) && (code_pos == 0)) ||
         ((argc == apos + 4) && (code_pos != 0))) {
-        gcoap_req_init(&pdu, &buf[0], CONFIG_GCOAP_PDU_BUF_SIZE, code_pos+1, argv[apos+2]);
+
+        char *uri = argv[apos+2];
+        int uri_len = strlen(argv[apos+2]);
+
+        if (_proxied) {
+            uri_len = snprintf(proxy_uri, 64, "coap://[%s]:%s%s", argv[apos], argv[apos+1], uri);
+            uri = proxy_uri;
+        }
+
+        if (_proxied) {
+            gcoap_req_init(&pdu, &buf[0], CONFIG_GCOAP_PDU_BUF_SIZE, code_pos+1, NULL);
+            coap_opt_add_string(&pdu, COAP_OPT_PROXY_URI, uri, '\0');
```

Comment:
@kb2ma I added the helper function `coap_opt_put_proxy_uri()` to `nanocoap`, however, I don't think it's quite convenient to use in our scenario here (: since we do not have the `last option` at hand (without parsing). So, I used `coap_opt_add_string()`.

Reply:
`coap_opt_add_string()` is part of the Packet API for nanocoap, which uses a coap_pkt_t to add options and does not require the last option. So the convenience function should be `coap_opt_add_proxy_uri()`. In contrast `coap_opt_put_proxy_uri()` is part of the Buffer API and requires the last option. See the [nanocoap module](http://doc.riot-os.org/group__net__nanocoap.html) documentation to understand the distinction. Notice there are separate sections for the two APIs.

I created the Packet API to be a simpler alternative. :-)

Reply:
Apologies if I lead you astray be referencing `coap_opt_put_location_path()` as the model.

Reply:
Alright, thanks for the hint :) will have a look now and will then return with a patch :)

Reply:
@kb2ma how about now? I also added some remarks to the doc.